doc/services/device_mgmt/smp_groups/smp_group_0.rst

```diff
@@ -91,23 +91,44 @@ CBOR data of successful response:
 
 In case of error the CBOR data takes the form:
 
-.. code-block:: none
+.. tabs::
 
-    {
-        (str)"rc"       : (int)
-    }
+   .. group-tab:: SMP version 2
+
+      .. code-block:: none
+
+          {
+              (str)"err" : {
+                  (str)"group"    : (uint)
+                  (str)"rc"       : (uint)
+              }
+          }
+
+   .. group-tab:: SMP version 1
+
+      .. code-block:: none
+
+          {
+              (str)"rc"       : (int)
+          }
 
 where:
 
 .. table::
     :align: center
 
-    +-----------------------+-----------------------------------------------+
-    | "r"                   | Replying echo string                          |
-    +-----------------------+-----------------------------------------------+
-    | "rc"                  | :c:enum:`mcumgr_err_t`                        |
-    |                       | only appears if non-zero (error condition).   |
-    +-----------------------+-----------------------------------------------+
+    +------------------+-------------------------------------------------------------------------+
+    | "r"              | replying echo string.                                                   |
+    +------------------+-------------------------------------------------------------------------+
+    | "err" -> "group" | :c:enum:`mcumgr_group_t` group of the group-based error code. Only      |
+    |                  | appears if an error is returned when using SMP version 2.               |
+    +------------------+-------------------------------------------------------------------------+
+    | "err" -> "rc"    | contains the index of the group-based error code. Only appears if       |
+    |                  | non-zero (error condition) when using SMP version 2.                    |
+    +------------------+-------------------------------------------------------------------------+
+    | "rc"             | :c:enum:`mcumgr_err_t` only appears if non-zero (error condition) when  |
+    |                  | using SMP version 1.                                                    |
```

Comment:
How are the SMP errors returned with protocol version 2, they have own group now?

Reply:
Yes, "err" as the key then a map with "group" for group ID and "rc" for error code

Reply:
I mean the previous protocol error codes, like failed to alloc mem which could also be returned without getting into group specific code.

Reply:
They will still be returned as version 1 error codes

Reply:
I will clarify the text to show that better